java/query/src/main/java/sleeper/query/lambda/SqsQueryProcessorLambda.java

```diff
@@ -78,15 +78,10 @@ public Void handleRequest(SQSEvent event, Context context) {
 
         for (SQSEvent.SQSMessage message : event.getRecords()) {
             LOGGER.info("Received message with body {}", message.getBody());
-            Query query;
-            try {
-                query = serde.fromJson(message.getBody());
-                LOGGER.info("Deserialised message to query {}", query);
-            } catch (JsonParseException e) {
-                LOGGER.error("JSONParseException deserialsing query from JSON {}", message.getBody());
-                continue;
-            }
-            processor.processQuery(query);
+            event.getRecords().stream()
+                    .map(SQSEvent.SQSMessage::getBody)
+                    .flatMap(body -> queryValidator.deserialiseAndValidate(body).stream())
+                    .forEach(processor::processQuery);
```

Comment:
There's a problem here in that the processor and the validator don't agree on which query trackers should be used.

For example, a web socket query will never report validation failures to the web socket.